### DISCUSSION TITLE
Create canonical urls without additional parameter

### DISCUSSION THREAD

**Discussion by fnagel** [sulu/sulu#7605] [Q&A] [answered]
The SEO expert my client hired wants me to remove additional GET parameter from the canonical URLs. 

That means that adding some parameter to an URL should not be used within the generated canonical:
`<link rel="canonical" href="https://domain.com/path">` instead of e.g. `<link rel="canonical" href="https://domain.com/path?foo=bar">`

This is generated in the SEO twig partial `vendor/sulu/sulu/src/Sulu/Bundle/WebsiteBundle/Resources/views/Extension/seo.html.twig` by using `<link rel="canonical" href="{{ seoCanonical|default(app.request.uri) }}">` as `app.request.uri` always contains the full URL.

I've tried something like `sulu_content_path(app.request.pathInfo)` or `sulu_content_path(request.resourceLocator)` and both seems to work fine for Sulu pages and my custom controller pages but I'm unsure what is the best option to use. 

Can somebody help me out here?

**Comment by alexander-schranz** [comment, marked as answer]
The `sulu_content_path` adds the prefix for resourceLocator. So it is not good to use it with `pathInfo`. E.g. your pathInfo is `/en/test` and your webspace is `{host}/{localization}` it will end up in `.../en/en/test`

The best way to achieve what you want is to built directly ontop of Symfony `app.request` this works for all your routes even non sulu / symfony static routes:

```twig
{%- block canonical -%}
    {%- set seoCanonical = seoCanonical|default(absolute_url(app.request.pathInfo)) -%}
    {#- or -#}
    {%- set seoCanonical = seoCanonical|default(app.request.schemeAndHttpHost ~ app.request.pathInfo) -%}
    <link rel="canonical" href="{{ seoCanonical }}">
{%- endblock -%}
```

**Reply by fnagel** [reply]
Thanks for your answer Alexander! 

I'm now using 

```
{%- block canonical -%}
     Remove additional parameter 
    <link rel="canonical" href="{{ seoCanonical|default(absolute_url(app.request.pathInfo)) }}">
{%- endblock -%}
```

and it works just for fine for Sulu pages, custom controller and error pages.
